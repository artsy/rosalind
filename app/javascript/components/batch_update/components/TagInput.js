import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { colors } from './Layout'

export const PENDING = {
  ADD: 'ADD',
  REMOVE: 'REMOVE',
}

class TagInput extends React.Component {
  render() {
    const { name, pendingAction, onClick } = this.props
    return (
      <Capsule pendingAction={pendingAction} onClick={onClick}>
        {pendingAction === PENDING.REMOVE ? <strike>{name}</strike> : name}
      </Capsule>
    )
  }
}

TagInput.propTypes = {
  name: PropTypes.string.isRequired,
  pendingAction: PropTypes.string,
  onClick: PropTypes.func,
}

const Capsule = styled.div`
  ${props => {
    if (props.pendingAction === PENDING.REMOVE) {
      return `
        border: solid 1px ${colors.red};
        background: ${colors.redLightest};
      `
    } else if (props.pendingAction === PENDING.ADD) {
      return `
        border: solid 1px ${colors.purple};
        background: ${colors.purpleLightest};
      `
    } else {
      return `
        border: solid 1px ${colors.grayLighter};
        background: ${colors.grayLightest};
      `
    }
  }}
  border-radius: 5px;
  cursor: pointer;
  display: inline-block;
  margin-bottom: 20px;
  margin-right: 20px;
  padding: 10px;
  user-select: none;
`

export default TagInput
