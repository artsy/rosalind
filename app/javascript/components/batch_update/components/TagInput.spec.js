import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { mount } from 'enzyme'
import TagInput, { PENDING } from './TagInput'

it('renders a to-ignore tag correctly', () => {
  const rendered = renderer.create(
    <TagInput name="Hot Dog" pendingAction={null} />
  )
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders a to-remove tag correctly', () => {
  const rendered = renderer.create(
    <TagInput name="Hot Dog" pendingAction={PENDING.REMOVE} />
  )
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders a to-add tag correctly', () => {
  const rendered = renderer.create(
    <TagInput name="Hot Dog" pendingAction={PENDING.ADD} />
  )
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('accepts changes to the pending action via click events', () => {
  const changeHandler = jest.fn()
  const wrapper = mount(
    <TagInput name="Hot Dog" pendingAction={null} onClick={changeHandler} />
  )
  wrapper.find('div').simulate('click')
  expect(changeHandler.mock.calls.length).toEqual(1)
})
