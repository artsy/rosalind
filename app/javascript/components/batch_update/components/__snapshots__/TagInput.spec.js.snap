// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a to-add tag correctly 1`] = `
.c0 {
  border: solid 1px #6E1FFF;
  background: #F7F3FF;
  border-radius: 5px;
  cursor: pointer;
  display: inline-block;
  margin-bottom: 20px;
  margin-right: 20px;
  padding: 10px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  className="c0"
  onClick={undefined}
>
  Hot Dog
</div>
`;

exports[`renders a to-ignore tag correctly 1`] = `
.c0 {
  border: solid 1px #E5E5E5;
  background: #F8F8F8;
  border-radius: 5px;
  cursor: pointer;
  display: inline-block;
  margin-bottom: 20px;
  margin-right: 20px;
  padding: 10px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  className="c0"
  onClick={undefined}
>
  Hot Dog
</div>
`;

exports[`renders a to-remove tag correctly 1`] = `
.c0 {
  border: solid 1px #F7625A;
  background: #FFF8F8;
  border-radius: 5px;
  cursor: pointer;
  display: inline-block;
  margin-bottom: 20px;
  margin-right: 20px;
  padding: 10px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  className="c0"
  onClick={undefined}
>
  <strike>
    Hot Dog
  </strike>
</div>
`;
